### DISCUSSION TITLE
pocketbase as a framework hosting on fly.io

### DISCUSSION THREAD

**Discussion by dindinet** [pocketbase/pocketbase#1864] [Q&A] [answered]
Hi All.

Has anyone had success with using pocketbase as a framework and hosting on fly.io?

I created a go executable using pocketbase as a framework. It runs without problem on dev machine.
I have been trying to deploy to fly.io using the method outlined here https://github.com/pocketbase/pocketbase/discussions/537 substituting the references to pocketbase executable with reference to my executable

from fly.io log
===========
2023-02-16T18:19:51.565 app[8a4dcda1] lhr [info] Preparing to run: `./pb/fwpb serve --http=0.0.0.0:8080` as root
2023-02-16T18:19:51.568 app[8a4dcda1] lhr [info] Error: UnhandledIoError(Os { code: 2, kind: NotFound, message: "No such file or directory" })

**Comment by ganigeorgiev** [comment, marked as answer]
Could you share your Dockerfile?

The process shouldn't be much different since as far as I understand you are not building the go executable inside the image, but attaching a locally prebuilt one, right?

If that's the case make sure that the file is copied in the resulting image (either via the `COPY` instruction or mounted as volume).

**Reply by clarity-strategies** [reply]
hi ganigeorgiev

Here is the Dockerfile 
**NOTE** the zip file contains the go binary and the 2 folders **pb_data** with a txt file and **pb_public** with index.html
![Screenshot from 2023-02-16 23-01-29](https://user-images.githubusercontent.com/17671620/219507779-e4d68a5e-fbdf-45fd-b82f-044600d352af.png)

    FROM alpine:latest

    RUN apk add --no-cache \
    unzip \
    openssh

    # download and unzip fwpb
    ADD https://clarity-strategies.github.io/fwpocketbase/fwpb.zip /tmp/fwpb.zip
    RUN unzip /tmp/fwpb.zip -d /pb/

    EXPOSE 8080

    # start fwpb
    CMD ["/pb/fwpb", "serve", "--http=0.0.0.0:8080"]

And Also the fly.toml

# fly.toml file generated for fwpb on 2023-02-16T15:53:27Z
    
    app = "fwpb"
    kill_signal = "SIGINT"
    kill_timeout = 5
    processes = []
    
    [env]
    
    [experimental]
      auto_rollback = true
    
    [mounts]
      source = "fwpb_vol"
      destination = "/pb"
    
    [[services]]
      http_checks = []
      internal_port = 8080
      processes = ["app"]
      protocol = "tcp"
      script_checks = []
      [services.concurrency]
        hard_limit = 25
        soft_limit = 20
        type = "connections"
    
      [[services.ports]]
        force_https = true
        handlers = ["http"]
        port = 80
    
      [[services.ports]]
        handlers = ["tls", "http"]
        port = 443
    
      [[services.tcp_checks]]
        grace_period = "1s"
        interval = "15s"
        restart_limit = 0
        timeout = "2s"

**Reply by etmartinkazoo** [reply]
This may be of help. I have a lot of production Pocketbase instances on Fly.io which has been phenomenal so far: [https://cloud.brem.io/Qysm5bXf](https://cloud.brem.io/Qysm5bXf)

**Reply by aluedtke7** [reply]
When you compile your binary, it's important to use this command:
`GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build`
Otherwise, the binary won't work in the alpine based container you're probably using.
Or have you tried to run the container locally on your dev machine?

**Reply by ganigeorgiev** [reply]
@clarity-strategies, @dindinet The mounted volume configuration doesn't seem correct. The `destination` path seems that it will replace the location of the extracted files. Or in other words, try to change your fly.toml `[mounts]` section to:

```toml
[mounts]
  source = "fwpb_vol"
  destination = "/pb/pb_data"
```

**Comment by dindinet** [comment]
@aluedtke7 @ganigeorgiev thankyou both your suggestions were required. I am now in pocketbase as a framework heaven.
@etmartinkazoo I will be sharing your video as often as I can to spread the pocketbase power.
